fms/utils/serialization.py

```diff
@@ -1,20 +1,42 @@
+import collections
+import itertools
 import os
-from collections import ChainMap, OrderedDict
+from collections import ChainMap
+from collections.abc import Iterable
 from pathlib import Path
-from typing import Any, Callable, List, Mapping, MutableMapping, Optional, Union
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Tuple,
+    Union,
+)
 
 import torch
 
+from fms.modules.tp import TPModule
+
 
 __adapters: MutableMapping[str, MutableMapping[str, Callable[[Mapping], Mapping]]] = {}
+__fusable_weight_groups: MutableMapping[
+    str, MutableMapping[str, Callable[[List[str]], List[Union[str, List[str]]]]]
+] = {}
 
 
 def register_adapter(
-    architecture: str, source: str, adapter: Callable[[Mapping], Mapping]
+    architecture: str,
+    source: str,
+    adapter: Callable[[Mapping], Mapping],
+    fwg_adapter: Optional[Callable[[List[str]], List[Union[str, List[str]]]]] = None,
```

Comment:
this seems like it will be difficult for a user of the API to understand or configure, and I don't see an example usage or comment for the parameter

Reply:
the issue is that we don't really need this until we are loading from checkpoints with unfused q,k,v into models with a fused qkv matrix. I'll add the docs (I forgot), but I was planning to have an example as part of the PR for fused qkv that @JRosenkranz was planning to do